### PR TITLE
feat: Save and Loads Nodes From Disk

### PR DESCRIPTION
**Pull request by blessedcoolant**

### CODE REVIEW

**Comment by mickr777** [comment]
@blessedcoolant can we add the the fitview for refresh or changing tabs back to nodes, I think it being this way is better then now when you change to nodes tab it zoom right in

```
  const onInit: OnInit = useCallback(
    (v) => {
      dispatch(setEditorInstance(v));
      if (v && typeof v.fitView === 'function') {
        v.fitView();
      }
    },
    [dispatch]
  );
```

**Comment by blessedcoolant** [comment]
@mickr777 Done.

**Comment by mickr777** [comment]
@blessedcoolant should there be a clear/reset button with a confirm you want to clear all? eg if I load a large layout, then I want to start a new fresh one I have to go and delete every node

**Comment by blessedcoolant** [comment]
> @blessedcoolant should there be a clear/reset button with a confirm you want to clear all? eg if I load a large layout, then I want to start a new fresh one I have to go and delete every node

There should we. We'll add one. I'll need to setup the modals for double checking. You can also shfit drag to select all nodes and back space to delete them all at once for the time being.

**Comment by mickr777** [comment]
@blessedcoolant I know that these things are highly unimportant but, while we had this pr open, do you think we should tidy up the reload schema button
image

**Comment by psychedelicious** [comment]
I'm not 100% sure this is the right approach.

We won't be able to save this object to images for sharing. Images will have the same graph as the backend, so they will need separate logic to parse and load a graph from metadata.

What happens when the `reactflow` API changes? Your old saved serialized `editorInstance`s are no longer valid. What if we change our own data format? No longer valid.

We may want to establish our own format that enriches the backend graph format with potential for future things like named edges, groups, etc.

In other words, we want to be able to derive a valid node editor state from the saved graph. Saving the exact node editor state and naively shoving it into `reactflow` is highly unlikely to be maintainable.

**Review by psychedelicious — CHANGES_REQUESTED** [review]
Please separate the changes unrelated to graph saving and loading into their own PR, which I'm happy to approve.

Also please share a ComfyUI saved graph / image-with-graph-embedded so I can review that. Curious what all they are storing.

**Review by psychedelicious — APPROVED** [review]
